src/ecdsa/keys.py

```diff
@@ -1133,3 +1133,22 @@ def sign_number(self, number, entropy=None, k=None):
         assert 1 <= _k < order
         sig = self.privkey.sign(number, _k)
         return sig.r, sig.s
+
+    def ecdh_get_shared_secret(self, remote_public_key):
```

Comment:
Like I said, I'd rather not have ECDH as part of VerifyingKey or SigningKey, if you'd prefer to work on this in steps, then I think this should be a standalone method in a new module (`ecdh.py`) and operate on `ecdsa.ecdsa.Public_key` and `ecdsa.ecdsa.Private_key` objects or a method of the `Private_key` class (`derive` or something)

Reply:
my worry is API stability: SigningKey is part of the public API. If we release a version with this method included we can't remove it without breaking the public API